src/main/java/servlets/DataEntries.java

```diff
@@ -33,14 +35,8 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       }
     }
     // get JSON from Database.
-    try {
-      Database database = new GCSDatabase(sharedObjects.databaseName);
-      String json = DatabaseQuery.getJson(database);
-      response.getWriter().println(json);
-    } catch (Exception e) {
-      System.out.println("*************COULD NOT INITIALIZE DATABASE*********************");
-      System.out.println("Exception while initializing" + e.getMessage());
-      throw new RuntimeException(e.getMessage());
-    }
+    database = new GCSDatabase(sharedObjects.databaseName);
```

Comment:
The `new GCSDatabase(...)` code will run on every call to `doGet`, so you're constructing a new database object on every GET request. I think you can override `GenericServlet.init` and do the construction there instead, so that you only create the database once instead of in every request.

https://javaee.github.io/javaee-spec/javadocs/javax/servlet/GenericServlet.html#init--